docs/CLASS_TABLE.md

```diff
@@ -0,0 +1,655 @@
+
+</td>
+</tr>
+<tr>
+<td>
+
+`Control.Applicative (Applicative)`, `Prelude (Applicative)`
+
+```haskell
+class Functor f => Applicative f where
+  pure :: a -> f a
+  (<*>) :: f (a -> b) -> f a -> f b
+  liftA2 :: (a -> b -> c) -> f a -> f b -> f c
+  (*>) :: f a -> f b -> f b
+  (<*) :: f a -> f b -> f a
+```
+
+</td>
+<td>
+
+[**Data Applicative**](https://www.tweag.io/blog/2020-01-16-data-vs-control#data-functors)
+
+`Data.Functor.Linear (Applicative)`
+
+```haskell
+class Data.Functor f => Data.Applicative f where
+  pure :: a -> f a
+  (<*>) :: f (a %1 -> b) %1 -> f a %1 -> f b
+  liftA2 :: (a %1 -> b %1 -> c) -> f a %1 -> f b %1 -> f c
+```
+
+N.B.: a linear Data variant of `(<*)` is available as a function (not method) in `Prelude.Linear` with the following signature:
+
+```haskell
+(<*) :: (Data.Applicative f, Consumable b) => f a %1 -> f b %1 -> f a
+```
+
+`(*>)` is not available though.
+
+[**Control Applicative**](https://www.tweag.io/blog/2020-01-16-data-vs-control#control-functors)
+
+`Control.Functor.Linear (Applicative)`
+
+```haskell
+class (Data.Applicative f, Control.Functor f) => Control.Applicative f where
+  pure :: a %1 -> f a
+  (<*>) :: f (a %1 -> b) %1 -> f a %1 -> f b
+  liftA2 :: (a %1 -> b %1 -> c) %1 -> f a %1 -> f b %1 -> f c
+```
+
+N.B.: linear Control variants of `(<*)` and `(*>)` are not available.
+
+</td>
+</tr>
+<tr>
+<td>
+
+`Control.Monad (Monad)`, `Prelude (Monad)`
+
+```haskell
+class Applicative m => Monad m where
+  (>>=) :: forall a b. m a -> (a -> m b) -> m b
+  (>>) :: forall a b. m a -> m b -> m b
+  return :: a -> m a
+```
+
+</td>
+<td>
+
+`Control.Functor.Linear (Monad)`
+
+```haskell
+class Control.Applicative m => Monad m where
+  (>>=) :: m a %1 -> (a %1 -> m b) %1 -> m b
+  (>>) :: m () %1 -> m a %1 -> m a
+```
+
+N.B.: `return` is available as a function (not method) in the same module.
+
+</td>
+</tr>
+<tr>
+<td>
+
+`Control.Monad.Fail (MonadFail)`, `Prelude (MonadFail)`
+
+```haskell
+class Monad m => MonadFail m where
+  fail :: String -> m a
+```
+
+</td>
+<td>
+
+`Control.Functor.Linear (MonadFail)`
+
+```haskell
+class Monad m => MonadFail m where
+  fail :: String -> m a
+```
+
+</td>
+</tr>
+<tr>
+<td>
+
+`Data.Foldable (Foldable)`, `Prelude (Foldable)`
+
+```haskell
+class Foldable t where
+  fold :: Monoid m => t m -> m
+  foldMap :: Monoid m => (a -> m) -> t a -> m
+  foldMap' :: Monoid m => (a -> m) -> t a -> m
+  foldr :: (a -> b -> b) -> b -> t a -> b
+  foldr' :: (a -> b -> b) -> b -> t a -> b
+  foldl :: (b -> a -> b) -> b -> t a -> b
+  foldl' :: (b -> a -> b) -> b -> t a -> b
+  foldr1 :: (a -> a -> a) -> t a -> a
+  foldl1 :: (a -> a -> a) -> t a -> a
+  toList :: t a -> [a]
+  null :: t a -> Bool
+  length :: t a -> Int
+  elem :: Eq a => a -> t a -> Bool
+  maximum :: forall a. Ord a => t a -> a
+  minimum :: forall a. Ord a => t a -> a
+  sum :: Num a => t a -> a
+  product :: Num a => t a -> a
+```
+
+</td>
+<td>
+
+No linear counterpart available at the moment.
```

Comment:
Perhaps mention the folds in `Data.List.Linear` here? That would have helped me.

Reply:
Good idea!

Reply:
```suggestion
No linear counterpart available at the moment, but individual fold functions can be found in `Data.List.Linear`
```